sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSourceResolution.scala

```diff
@@ -173,6 +173,19 @@ case class DataSourceResolution(
       // only top-level adds are supported using AlterTableAddColumnsCommand
       AlterTableAddColumnsCommand(table, newColumns.map(convertToStructField))
 
+    case DeleteFromStatement(AsTableIdentifier(table), tableAlias, condition) =>
+      throw new AnalysisException(
```

Comment:
Since this always throws `AnalysisException`, I think this case should be removed. Instead, the next case should match and the `V2SessionCatalog` should be used. If the table loaded by the v2 session catalog doesn't support delete, then conversion to physical plan will fail when `asDeletable` is called.

Then users can still call v2 deletes for formats like `parquet` that have a v2 implementation that will work.

FYI @brkyvz.

Reply:
Thank you @rdblue . Removed this case and fallback to `sessionCatalog` when `resolveTables` for `DeleteFromTable`.

Reply:
I think it's worse to move this case from here to https://github.com/apache/spark/pull/25115/files#diff-57b3d87be744b7d79a9beacf8e5e5eb2R657 .

If we can't merge these 2 cases into one here, let's keep it as it was.

Reply:
If I understand correctly, one purpose of removing the first case is we can execute delete on `parquet` format via this API (if we implement it later) as @rdblue mentioned. The key point here is we resolve the table use `V2SessionCatalog` **as the fallback catalog**.  The original `resolveTable` doesn't give any fallback-to-sessionCatalog mechanism (if no catalog found, it will fallback to `resolveRelation`). So maybe we can modify `resolveTable` and let it treat `V2SessionCatalog` as a try option:
```
case u @ UnresolvedRelation(nameParts) =>
        val CatalogObjectIdentifier(maybeCatalog, ident) = nameParts
        val catalog = maybeCatalog.orElse(sessionCatalog)
        catalog match {
          case Some(catalogPlugin) =>
            loadTable(catalogPlugin, ident).map(DataSourceV2Relation.create).getOrElse(u)
          case None =>
            u
        }
```

Reply:
I don't think we need to update `ResolveTables`, though I do see that it would be nice to use `ResolveTables` as the only rule that resolves `UnresolvedRelation` for v2 tables.

There is already another rule that loads tables from a catalog, [`ResolveInsertInto`](https://github.com/apache/spark/blob/master/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala#L772).

I considered updating that rule and moving the table resolution part into `ResolveTables` as well, but I think it is a little cleaner to resolve the table when converting the statement (in `DataSourceResolution`), as @cloud-fan is suggesting.

One of the reasons to do this for the insert plans is that those plans don't include the target relation as a child. Instead, those plans have the data to insert as a child node, which means that the unresolved relation won't be visible to the `ResolveTables` rule.

Taking the same approach in this PR would also make this a little cleaner. If `DeleteFrom` didn't expose the relation as a child, it could be a `UnaryNode` and you wouldn't need to update some of the other rules to explicitly include `DeleteFrom`.

Reply:
Okay, I rolled back the resolve rules for `DeleteFromTable` as it was as @cloud-fan suggested. For cases that like deleting from formats or `V2SessionCatalog` support, let's open another pr. And another pr for resolve rules is also need because I found other issues related with that. Does this sounds reasonable?

Reply:
We can remove this case after https://github.com/apache/spark/pull/25402, which updates `ResolveTable` to fallback to v2 session catalog.

Reply:
Saw the code in #25402 . I think it's the best choice.